File: 978-valid-mountain-array/valid-mountain-array.ts
function validMountainArray(arr: number[]): boolean {
    let left = 0, right = arr.length - 1;
    while (arr[left] < arr[left + 1] || arr[right] < arr[right - 1]) {
        if (arr[left] < arr[left + 1]) {
            left += 1;
        }
        if (arr[right] < arr[right - 1]) {
            right -= 1;
        }
    }
    if (left !== right || left === arr.length - 1 || right === 0) {
        return false;
    }
    return true;
};

File: 978-valid-mountain-array/valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        if (arr.length < 3) {
            return false;
        }
        int start = 0;
        int end = arr.length - 1;
        while (start < end) {
            if (arr[start + 1] > arr[start]) {
                start += 1;
            } else if (arr[end - 1] > arr[end]) {
                end -= 1;
            } else {
                break;
            }
        }
        return start != 0 && end != arr.length - 1 && start == end;
    }
}